### PR TITLE
Enhanced runtests.sh and reenabled another block of tests

### PR DESCRIPTION
**Pull request by dmatej**
### Targets

- improve code coverage with anything usable we have
- have control over JEE10 - no new bugs, rather fixing old ones
- prepare a base for future modernisation (these tests should pass first then I can rewrite them - lighter, faster, usable)

### Changes in Tests

- runtests.sh is able to resolve the glassfish version automatically
- runtests.sh now respects also already existing bundles/glassfish.zip
- runtests.sh and gftest.sh now respect also already set env variables
- some ports can be reset by env properties (but it still doesn't work everywhere - limits usage on local machines)
- fixed port evaluation
- fixed JDK17 compatibility - implicit java.lang.Record conflicts with javax.resource.cci.Record
- fixed naming_all, deployment_all, security_all, webservice_all - mostly obsoleted descriptors
- removed old NetBeans files
- removed dead tests using obsoleted API - JAX-RPC, Managed Beans

### Fixed bugs

- NPE in WebServiceReferenceManagerImpl.java
- Signing jar files, broken in 18f78207a194e756b49c1fb25bdcdf5f1b570b0e - the JarSigner wraps the stream too.
- signJar tried to delete the file before it was closed.
- fixed error message for the signJar - '{0}' is the original file, not target
- fixed appclient classpath - defined classloader must be available (broken maybe by 93938a820adaa28edbe18cd26eafe3bf3eb0e7c8, but it didn't work before that too)
- Added add-opens for java.io - fixed naming_all tests which used cluster instances

### CODE REVIEW

File: appserver/webservices/jsr109-impl/src/main/java/org/glassfish/webservices/WebServiceReferenceManagerImpl.java [hunk 1]
```diff
@@ -473,9 +481,13 @@ private ServletAdapter getServletAdapterBasedOnWsdlUrl(ServiceReferenceDescripto
         }
 
         URL wsdl = desc.getWsdlFileUrl();
+        if (wsdl == null) {
```
Comment (dmatej) [inline appserver/webservices/jsr109-impl/src/main/java/org/glassfish/webservices/WebServiceReferenceManagerImpl.java:484]: wsdl now can be null - some of tests caused NPE on the following line.

File: appserver/appclient/server/core/src/main/java/org/glassfish/appclient/server/core/jws/servedcontent/ASJarSigner.java [hunk 2]
```diff
@@ -66,122 +70,110 @@
...
 
-    /** keystore type for JKS keystores */
-    private static final String JKS_KEYSTORE_TYPE_VALUE = "jks";
-
     /** default alias for signing if the admin does not specify one */
     private static final String DEFAULT_ALIAS_VALUE = "s1as";
 
-    private static final String DEFAULT_DIGEST_ALGORITHM = "SHA-1";
-    private static final String DEFAULT_SIGNATURE_ALGORITHM = "SHA1withRSA";
+    private static final String DEFAULT_DIGEST_ALGORITHM = "SHA-256";
```
Comment (dmatej) [inline appserver/appclient/server/core/src/main/java/org/glassfish/appclient/server/core/jws/servedcontent/ASJarSigner.java:76]: Changed to standards of these days.

File: appserver/appclient/server/core/src/main/java/org/glassfish/appclient/server/core/jws/servedcontent/ASJarSigner.java [hunk 3]
```diff
@@ -66,122 +70,110 @@
...
+     * @param unsignedJar the unsigned JAR file
+     * @param signedJar the signed JAR to be created
+     * @param attrs additional attributes to be added to the JAR's manifest main section
+     * @return the elapsed time to sign the JAR (in milliseconds)
+     * @throws Exception getting the keystores from SSLUtils fails
      */
-    public long signJar(final File unsignedJar, final File signedJar,
-        String alias, Attributes attrs) throws Exception {
-
-        final ZipOutputStream zout = new ZipOutputStream(
```
Comment (dmatej) [inline appserver/appclient/server/core/src/main/java/org/glassfish/appclient/server/core/jws/servedcontent/ASJarSigner.java:103]: BUG caused in 18f78207a194e756b49c1fb25bdcdf5f1b570b0e - JarSigner wraps the stream too

File: appserver/appclient/server/core/src/main/java/org/glassfish/appclient/server/core/jws/servedcontent/ASJarSigner.java [hunk 4]
```diff
@@ -66,122 +70,110 @@
...
+    public long signJar(final File unsignedJar, final OutputStream signedJar, String alias, final Attributes attrs,
+        final Map<String, byte[]> additionalContent) throws IllegalStateException {
         if (alias == null) {
             alias = DEFAULT_ALIAS_VALUE;
         }
         long startTime = System.currentTimeMillis();
         long duration = 0;
         synchronized(this) {
             try {
-                Certificate[] certificates = null;
```
Comment (dmatej) [inline appserver/appclient/server/core/src/main/java/org/glassfish/appclient/server/core/jws/servedcontent/ASJarSigner.java:143]: Just refactoring - extracted to own mehod

File: appserver/appclient/server/core/src/main/java/org/glassfish/appclient/server/core/jws/servedcontent/AutoSignedContent.java [hunk 5]
```diff
@@ -151,7 +148,15 @@ private void createSignedFile(final URI requestURI) throws Exception {
             }
         }
         final Attributes attrs = createJWSAttrs(requestURI, appName);
-        jarSigner.signJar(unsignedFile, signedFile, userProvidedAlias, attrs);
```
Comment (dmatej) [inline appserver/appclient/server/core/src/main/java/org/glassfish/appclient/server/core/jws/servedcontent/AutoSignedContent.java:154]: signJar tried to delete the file before it was closed.

File: appserver/appclient/server/core/src/main/resources/org/glassfish/appclient/server/core/jws/servedcontent/LocalStrings.properties [hunk 6]
```diff
@@ -22,7 +22,7 @@ jws.sign.noUnsignedJar=Unsigned jar {0} marked for signing does not exist
 jws.sign.userAliasAbsent=User-specified alias {0} to be used for signing JAR files was not found in any keystore; attempting to use default alias instead
 jws.sign.aliasNotFound=Alias {0} not found in keystore(s); aborting
 jws.sign.aliasFoundMult=Alias {0} to be used for signing JAR files was found in more than one keystore; using the first occurrence from the keystores with these providers: {1}
-jws.sign.errorSigning=Error attempting to create signed jar {0} with alias {1}
```
Comment (dmatej) [inline appserver/appclient/server/core/src/main/resources/org/glassfish/appclient/server/core/jws/servedcontent/LocalStrings.properties:25]: In fact the `{0}` is the original file.

**Review by arjantijms — APPROVED** [review]

**Comment by arjantijms** [comment]
Wow, that's an amazing piece of work again. Thanks!

File: appserver/appclient/server/core/src/main/java/org/glassfish/appclient/server/core/jws/servedcontent/ASJarSigner.java (hunk 3 above)
Comment (dmatej) [reply appserver/appclient/server/core/src/main/java/org/glassfish/appclient/server/core/jws/servedcontent/ASJarSigner.java:103]: Just for the record - the error message  from the JDK is not much helpful - "there is no current entry". But it is correct - there was no current entry for adding an entry into a zip file, which would be another zipfile.

**Pull request merged**